core/src/main/java/org/hisp/dhis/android/core/arch/db/querybuilders/internal/SQLStatementBuilder.kt

```diff
@@ -25,38 +25,29 @@
  *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-
-package org.hisp.dhis.android.core.arch.db.querybuilders.internal;
-
-import org.hisp.dhis.android.core.arch.db.stores.projections.internal.LinkTableChildProjection;
-
-public interface SQLStatementBuilder extends ReadOnlySQLStatementBuilder {
-
-    String getTableName();
-
-    String[] getColumns();
-
-    String selectUids();
-
-    String selectUidsWhere(String whereClause);
-
-    String selectUidsWhere(String whereClause, String orderByClause);
-
-    String selectColumnWhere(String column, String whereClause);
-
-    String selectChildrenWithLinkTable(LinkTableChildProjection projection, String parentUid, String whereClause);
-
-    String selectByUid();
-
-    String selectDistinct(String column);
-
-    String insert();
-
-    String update();
-
-    String updateWhere();
-
-    String deleteById();
-
-    String deleteWhere();
+package org.hisp.dhis.android.core.arch.db.querybuilders.internal
+
+import org.hisp.dhis.android.core.arch.db.stores.projections.internal.LinkTableChildProjection
+
+@Suppress("TooManyFunctions")
+internal interface SQLStatementBuilder : ReadOnlySQLStatementBuilder {
+    fun getTableName(): String
+    fun getColumns(): Array<String>
+    fun selectUids(): String?
```

Comment:
Sorry, I didn't notice it before. This should return a non-null String. It is better to change it now to avoid cascading nulability errors when we migrate other parts of the code